services/thermal_mgr/thermal_mgr.c

```diff
@@ -42,19 +42,40 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
+  if(event == NULL) return ERR_CODE_INVALID_ARG;
+  if(thermalMgrQueueHandle == NULL) return ERR_CODE_INVALID_STATE;
+
+  if(xQueueSend(thermalMgrQueueHandle, event, 0) != pdTRUE) return ERR_CODE_QUEUE_FULL;
 
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+  thermal_mgr_event_t event = {.type = THERMAL_MGR_EVENT_ISR_OVERTEMP_CHECK};
+  thermalMgrSendEvent(&event);
+
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  lm75bd_config_t *config = (void *) pvParameters;
+  thermal_mgr_event_t event;
   while (1) {
-    
+    if (xQueueReceive(thermalMgrQueueHandle, &event, portMAX_DELAY) == pdTRUE) {
+      if (event.type != THERMAL_MGR_EVENT_MEASURE_TEMP_CMD 
+       && event.type != THERMAL_MGR_EVENT_ISR_OVERTEMP_CHECK){
+        continue; LOG_ERROR_CODE(ERR_CODE_INVALID_STATE);
+      }
+      
+      float temp;
+      readTempLM75BD(LM75BD_OBC_I2C_ADDR, &temp);
```

Comment:
check if the readTempLM75BD returns success, log if it does not. ideally by doing LOG_IF_ERROR_CODE, since readTempLM75BD returns a error_code_t. more details about LOG_IF_ERROR_CODE can be found in the onboarding challenge document

Reply:
generally, if you are calling a function returning an error code you should always check if its successful, and log if it isn't. this is very important for debugging